File: app/src/main/java/com/hqq/core/app/ui/recycle/BannerActivity.java
package com.hqq.core.app.ui.recycle;

import com.hqq.core.recycler.RecycleViewBanner;
import com.hqq.core.ui.BaseActivity;
import com.hqq.core.app.R;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.coreapp.ui.activity
 * @FileName :   BannerActivity
 * @Date : 2018/12/20 0020
 * @Email :  qiqiang213@gmail.com
 * @Descrive : TODO
 */
public class BannerActivity extends BaseActivity {

    @BindView(R.id.rc_banner)
    RecycleViewBanner mRcBanner;
    @BindView(R.id.rc_banner2)
    RecycleViewBanner mRcBanner2;

    @Override
    public int setViewId() {
        return R.layout.activity_banner;
    }


    @Override
    public void initView() {
        List<String> list = new ArrayList<>();
        list.add(("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg"));
        list.add(("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg"));
        list.add(("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg"));
        list.add(("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg"));
        list.add(("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg"));
        mRcBanner.setRvBannerData(list);
        List<Integer> list2 = new ArrayList<>();
        list2.add(R.mipmap.ic_banner2);
        list2.add(R.mipmap.ic_banner2);
        mRcBanner2.setRvBannerData(list2);
    }

}

File: app/src/main/java/com/easy/example/ui/recycle/BannerActivity.kt
package com.easy.example.ui.recycle

import com.easy.core.recycle.RecycleViewBanner
import com.easy.core.ui.base.BaseActivity
import com.easy.core.ui.base.BaseViewBindingActivity
import com.easy.example.R
import com.easy.example.databinding.ActivityBannerBinding

/**
 * @Author : huangqiqiang
 * @Package : com.easy.coreapp.ui.activity
 * @FileName :   BannerActivity
 * @Date : 2018/12/20 0020
 * @Email :  qiqiang213@gmail.com
 * @Describe : TODO
 */
class BannerActivity : BaseViewBindingActivity<ActivityBannerBinding>() {
    var mRcBanner: RecycleViewBanner? = null
    var mRcBanner2: RecycleViewBanner? = null


    override fun initView() {
        mRcBanner = findViewById(R.id.rc_banner)
        mRcBanner2 = findViewById(R.id.rc_banner2)
        val list = mutableListOf<Any>()
        list.add("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg")
        list.add("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg")
        list.add("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg")
        list.add("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg")
        list.add("http://pic2.sc.chinaz.com/files/pic/webjs1/201903/jiaoben6644.jpg")
        mRcBanner?.setRvBannerData(list)
        val list2: MutableList<Any> = ArrayList()
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        list2.add(R.mipmap.ic_banner2)
        mRcBanner2?.setRvBannerData(list2)



        binding.button100.setOnClickListener {
            mRcBanner2?.smoothScrollToPosition(100)
            mRcBanner?.scrollToPosition(100)
        }

    }
}